okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -344,8 +344,33 @@ private void waitTenantInit(Tenant tenant, ModuleInstance getInstance,
    * @return fut future
    */
   private Future<Void> invokeTenantInterface(Tenant tenant, TenantInstallOptions options,
-                                             ModuleDescriptor mdFrom, ModuleDescriptor mdTo,
-                                             ProxyContext pc) {
+      ModuleDescriptor mdFrom, ModuleDescriptor mdTo,
+      ProxyContext pc) {
+
+    Future<Void> future;
+    if (mdTo != null && options.getPurge()) {
+      // enable with purge is turned into purge on its own, followed by regular enable
+      mdFrom = null;
+      future = invokeTenantInterface1(tenant, options, mdTo, null, pc)
+          .otherwise(res -> {
+            logger.info("Tenant purge error for module {} ignored: {}",
+                mdTo.getId(), res.getMessage());
+            return null;
+          })
+          .map(x -> {
+            options.setPurge(false);
+            return null;
+          });
+    } else {
+      future = Future.succeededFuture();
+    }
+    final ModuleDescriptor mdFromFinal = mdFrom;
```

Comment:
Another possibility is to pass the mdFromFinal as Future success value.

Reply:
ok